server/src/main/java/com/jungleapp/cs414/server/Match.java

```diff
@@ -92,26 +94,32 @@ String forfeitMatch() {
 
     private void saveUpdatedMatch() {
         // TODO: Finish updating match with some sort of match identifier(gameID).
-//        try {
-//            Statement statement = connection.createStatement();
-//            String formattedTime = null;
-//            if (!isActive) {
-//                LocalDateTime matchEndTime = LocalDateTime.now();
-//                DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH-mm-ss");
-//
-//                formattedTime = matchEndTime.format(timeFormatter);
-//            }
-//
-//            //Update database entry with current details.
-//            statement.execute("UPDATE Game SET board='" + gson.toJson(this.board) + "'," +
-//                    "status='" + this.isActive + "'," +
-//                    "playerTurn='" + this.whoseTurn + "'," +
-//                    "winner='" + this.winner + "'," +
-//                    "endTime='" + formattedTime + "' WHERE TODO: Insert applicable match identifier here.");
-//
-//        } catch (SQLException e) {
-//            e.printStackTrace();
-//        }
+        try {
+            Statement statement = connection.createStatement();
+
+            if (currentMatch.status.equals("Finished")) {
+                LocalDateTime matchEndTime = LocalDateTime.now();
+                DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH-mm-ss");
+
+                String formattedTime = matchEndTime.format(timeFormatter);
+
+                statement.execute("UPDATE Game SET board='" + currentMatch.jungleBoard.getBoardJSON() + "'," +
+                        "status='" + currentMatch.status + "'," +
+                        "playerTurn='" + currentMatch.playerTurn + "'," +
```

Comment:
Should player turn be null for a finished game?

Reply:
That's a good point. It may help fix some of the turn errors too.